libraries/AP_Motors/AP_MotorsMatrix.cpp

```diff
@@ -246,9 +246,14 @@ void AP_MotorsMatrix::output_armed_stabilizing()
     // ensure that throttle_avg_max is between the input throttle and the maximum throttle
     throttle_avg_max = constrain_float(throttle_avg_max, throttle_thrust, throttle_thrust_max);
 
-    // throttle providing maximum roll, pitch and yaw range
-    // calculate the highest allowed average thrust that will provide maximum control range
-    float throttle_thrust_best_rpy = MIN(0.5f, throttle_avg_max);
+    // Ensure desired altitude output is not sacrificed for attitude output, but allow sacrifice of climb rate.
+    float desired_altitude_output = MIN(throttle_thrust, get_throttle_hover() * compensation_gain);
```

Comment:
> The motor mixer does not deal with altitude or have any knowledge of altitude. This name does not reflect what this variable is doing : desired_altitude_output

im open to changing the name, if you have a suggestion. but altitude change is inferred from the hover throttle. e.g. if throttle is above hover throttle, altitude increases, if below, decreases.

> There is nothing that ensures that limiting the throttle_thrust_best_rpy to be no less than hover throttle will prevent the aircraft from descending.

i have to strongly disagree here. `throttle_thrust_best_rpy` seems to be named quite clearly: throttle at best rpy. "best rpy" is basically the saturation limit. as the SITL tests show, at the saturation limit, altitude is lost, because throttle is `throttle_thrust_best_rpy`, and it is below hover throttle. i am not sure exactly how we disagree here. can you explain your thought process in more detail?